src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2342,7 +2342,8 @@ def _get_log(url, user_name, password, log_file=None):
                 # Extra encode() and decode for stdout which does not surpport 'utf-8'
                 logger.warning(chunk.decode(encoding='utf-8', errors='replace')
                                .encode(std_encoding, errors='replace')
-                               .decode(std_encoding, errors='replace'), end='')  # each line of log has CRLF.
+                               .decode(std_encoding, errors='replace')
+                               .rstrip('\n\r'))  # each line of log has CRLF.
```

Comment:
`.rstrip('\n\r')` might remove newlines which were intentionally logged by the app, so this is probably not ideal. But I don't know what line terminator is the extra one: `\n`, `\r`, `\r\n`?

From what I've seen, it seems to be `\n`. So replacing `rstrip` with `[:-1]` should do the job, as long as it is guaranteed that there will always be a trailing newline.